packages/scaffold/src/views/w3m-network-switch-view/index.ts

```diff
@@ -134,7 +134,12 @@ export class W3mNetworkSwitchView extends LitElement {
       this.error = false
       if (this.network) {
         await NetworkController.switchActiveNetwork(this.network)
-        if (!OptionsController.state.isSiweEnabled) {
+        if (OptionsController.state.isSiweEnabled) {
+          const { SIWEController } = await import('@web3modal/siwe')
+          if (SIWEController.state._client?.options?.signOutOnNetworkChange) {
+            await SIWEController.signOut()
+          }
+        } else {
```

Comment:
Should this logic live in `switchActiveNetwork` to avoid code duplication ?

Reply:
That causes a circular dependency cause it'd require `siwe` in `core` which is imported in `siwe` 😵 
But agree not cool to duplicate code

Reply:
abstracted to a utility 🙏